rhizome/host/lib/vm_setup.rb

```diff
@@ -770,4 +755,144 @@ def start_systemd_unit
   def restart_systemd_unit
     r "systemctl restart #{q_vm}"
   end
+
+  def build_ch_service(header:, footer:, slice_name:, mem_gib:, max_vcpus:, cpu_topology:, storage_volumes:, storage_params:, nics:, pci_devices:)
+    disk_params = storage_volumes.map { |volume|
+      if volume.read_only
+        "path=#{volume.image_path},readonly=on"
+      else
+        "vhost_user=true,socket=#{volume.vhost_sock},num_queues=#{volume.num_queues},queue_size=#{volume.queue_size}"
+      end
+    }
+    disk_params << "path=#{vp.cloudinit_img}"
+
+    disk_args =
+      if Gem::Version.new(@ch_version.version) >= Gem::Version.new("36")
+        "--disk #{disk_params.join(" ")}"
+      else
+        disk_params.map { |x| "--disk #{x}" }.join(" ")
+      end
+
+    net_params = nics.map { "--net mac=#{_1.mac},tap=#{_1.tap},ip=,mask=,num_queues=#{max_vcpus * 2 + 1}" }
+    pci_device_params =
+      if pci_devices.empty?
+        ""
+      elsif Gem::Version.new(@ch_version.version) >= Gem::Version.new("36")
+        "\\\n--device #{pci_devices.map { |dev| "path=/sys/bus/pci/devices/0000:#{dev[0]}/" }.join(" ")}"
+      else
+        "\\\n" + pci_devices.map { |dev| "--device path=/sys/bus/pci/devices/0000:#{dev[0]}/" }.join(" ")
+      end
+
+    <<~SERVICE
+  #{header}
+  [Service]
+  Slice=#{slice_name}
+  NetworkNamespacePath=/var/run/netns/#{@vm_name}
+  ExecStartPre=/usr/bin/rm -f #{vp.ch_api_sock}
+
+  ExecStart=#{@ch_version.bin} -v \\
+  --api-socket path=#{vp.ch_api_sock} \\
+  --kernel #{@firmware_version.path} \\
+  #{disk_args} \\
+  --console off --serial file=#{vp.serial_log} \\
+  --cpus boot=#{max_vcpus},topology=#{cpu_topology} \\
+  --memory size=#{mem_gib}G,#{@hugepages ? "hugepages=on,hugepage_size=1G" : "shared=on"} \\
+  #{net_params.join(" ")}
```

Comment:
Without a `\\` at the end of this line, is the next line still treated as part of `ExecStart`?